pkg/chunk/storage/metrics.go

```diff
@@ -0,0 +1,118 @@
+package storage
+
+import (
+	"context"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"github.com/weaveworks/common/user"
+
+	"github.com/cortexproject/cortex/pkg/chunk"
+)
+
+type metricsChunkClient struct {
```

Comment:
Sounds confusing. How about `chunkClientWithMetricsWrapper`? A bit verbose though.

Reply:
Given its limited scope, I think this works. Short comment would be nice :)